dask/dataframe/tests/test_multi.py

```diff
@@ -2077,8 +2077,10 @@ def test_merge_outer_empty():
 
     for x in range(0, k_clusters + 1):
         assert_eq(
-            dd.merge(empty_df, df[df.cluster == x], how="outer").compute(),
-            df[df.cluster == x].compute(),
+            dd.merge(empty_df, df[df.cluster == x], how="outer")
+            .compute()
```

Comment:
Do we really want the `compute` calls in this `assert_eq`?

Reply:
It looks like the real problem here is that _meta inference actually fails to produce the correct dtypes in `dd.merge(empty_df, df[df.cluster == x], how="outer")`, so the dtype check will fail until after compute.  The reason is that `merge` uses `_meta_nonempty` to infer the resulting dtypes.  However, for an empty DataFrame, the index dtype for `_meta_nonempty` will be random string values.  This results in NaN values in `meta = lhs._meta_nonempty.merge(rhs._meta_nonempty, **kwargs)` for an "outer" merge (causing "int" dtypes to become "float").

To get this particular test passing, we could avoid using `_meta_nonempty` for a collection that has no columns.

Reply:
Oh that is a nice solution! I didn't realize it would be more involved than removing the `compute()`. Can the other test lose its `compute()` now as well?

Reply:
I didn't check if I can yank other instances of `compute`, but that may be the case.  In some cases, I know that compute + sort_values is used to account for nondeterministic ordering in the result.

Reply:
Ah ok. Fine with me if that's the case here. Although ideally there'd be a kwarg on `assert_eq` that is like `check_order=False` :)

Reply:
I think there is that actually :open_mouth: !

Reply:
I'm not seeing the kwarg we want in the [assert_eq def](https://github.com/dask/dask/blob/34491c5b5c6252e59583bb55241f7b978cfd04b8/dask/dataframe/utils.py#L798)

Reply:
It gets passed along to the pandas one which has it though right?